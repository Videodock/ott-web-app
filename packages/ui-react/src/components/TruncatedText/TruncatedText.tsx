import React from 'react';
import classNames from 'classnames';

import styles from './TruncatedText.module.scss';

type TruncatedTextProps = {
  text: string;
  maximumLines: number;
  className?: string;
};

const TruncatedText: React.FC<TruncatedTextProps> = ({ text, maximumLines, className }) => {
  return (
    <div
      className={classNames(styles.truncatedText, className)}
      style={{
        maxHeight: `calc(1.5em * ${maximumLines})`,
        WebkitLineClamp: maximumLines,
      }}
    >
      {text}
    </div>
  );
};

export default TruncatedText;
