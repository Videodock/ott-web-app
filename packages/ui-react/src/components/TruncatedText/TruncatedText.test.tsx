import React from 'react';
import { render } from '@testing-library/react';

import TruncatedText from './TruncatedText';

describe('<TruncatedText>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<TruncatedText text="Test..." maximumLines={8} />);

    expect(container).toMatchSnapshot();
  });
});
