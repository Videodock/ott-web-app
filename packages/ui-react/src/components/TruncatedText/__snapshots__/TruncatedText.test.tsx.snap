// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TruncatedText> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_truncatedText_b1c064"
    style="max-height: calc(1.5em * 8); -webkit-line-clamp: 8;"
  >
    Test...
  </div>
</div>
`;
